openedx/core/djangoapps/ccxcon/tests/test_api.py

```diff
@@ -0,0 +1,188 @@
+
+@attr('shard_1')
+class APIsTestCase(SharedModuleStoreTestCase):
+    """
+    Unit tests for the API module functions
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    @classmethod
+    def setUpClass(cls):
+        super(APIsTestCase, cls).setUpClass()
+        cls.course = course = CourseFactory.create()
+        cls.course_key = cls.course.location.course_key
+
+        # Create a course outline
+        start = datetime.datetime(
+            2010, 5, 12, 2, 42, tzinfo=pytz.UTC
+        )
+        due = datetime.datetime(
+            2010, 7, 7, 0, 0, tzinfo=pytz.UTC
+        )
+
+        cls.chapters = [
+            ItemFactory.create(start=start, parent=course) for _ in xrange(2)
+        ]
+        cls.sequentials = flatten([
+            [
+                ItemFactory.create(parent=chapter) for _ in xrange(2)
+            ] for chapter in cls.chapters
+        ])
+        cls.verticals = flatten([
+            [
+                ItemFactory.create(
+                    start=start, due=due, parent=sequential, graded=True, format='Homework', category=u'vertical'
+                ) for _ in xrange(2)
+            ] for sequential in cls.sequentials
+        ])
+
+        # Trying to wrap the whole thing in a bulk operation fails because it
+        # doesn't find the parents. But we can at least wrap this part...
+        with cls.store.bulk_operations(course.id, emit_signals=False):
+            blocks = flatten([  # pylint: disable=unused-variable
+                [
+                    ItemFactory.create(parent=vertical) for _ in xrange(2)
+                ] for vertical in cls.verticals
+            ])
+
+    def setUp(self):
+        """
+        Set up tests
+        """
+        super(APIsTestCase, self).setUp()
+        # Create instructor account
+        self.instructor = AdminFactory.create()
+        # create an instance of modulestore
+        self.mstore = modulestore()
+        # enable ccx
+        self.course.enable_ccx = True
+        # setup CCX connector
+        self.course.ccx_connector = 'https://url.to.cxx.connector.mit.edu'
+        # save the changes
+        self.mstore.update_item(self.course, self.instructor.id)
+        # create a configuration for the ccx connector: this must match the one in the course
+        self.ccxcon_conf = CcxConFactory(url=self.course.ccx_connector)
+
+
+    def test_course_info_to_ccxcon_no_valid_course_key(self):
+        """
+        Test for an invalid course key
+        """
+        missing_course_key = CourseKey.from_string('course-v1:FakeOrganization+CN999+CR-FALL99')
+        self.assertIsNone(ccxconapi.course_info_to_ccxcon(missing_course_key))
+
+    def test_course_info_to_ccxcon_no_ccx_enabled(self):
+        """
+        Test for a course without CCX enabled
+        """
+        self.course.enable_ccx = False
+        self.mstore.update_item(self.course, self.instructor.id)
+        self.assertIsNone(ccxconapi.course_info_to_ccxcon(self.course_key))
```

Comment:
Since the function never returns anything anyways the assert against None isn't very useful. Is it possible to assert that the POST was not made, maybe through a mock?